# DATA EXPLORATION OPERATIONS USING PANDAS

In [1]:
import pandas as pd

In [2]:
sample=pd.read_csv('E:/cod-landings.csv')

##### We use pd.read_csv function to read the dataset in the PYTHON

## About The Dataset

##### The Atlantic and Pacific cod landings NOAA dataset is a collection of data on the fishing industry for Atlantic and Pacific cod from 1950 to 2021. It includes information on the year, state, and scientific name of the fish, as well as the pounds and metric tons of the catch and the dollars earned from the sale of the fish. The data in this dataset can be used to track the trends in cod fishing over time and to understand the economic impact of this industry in different states and regions. It can also be used to inform conservation and management efforts for these species.

In [3]:
sample

Year           State      NMFS Name       Pounds  Metric Tons  \
0     2021          ALASKA   COD, PACIFIC  330404171.0     149870.0   
1     2021     CONNECTICUT  COD, ATLANTIC       2277.0          1.0   
2     2021     CONNECTICUT  COD, ATLANTIC       7921.0          4.0   
3     2021        DELAWARE  COD, ATLANTIC         18.0          0.0   
4     2021           MAINE  COD, ATLANTIC       1171.0          1.0   
...    ...             ...            ...          ...          ...   
1191  1950  NORTH CAROLINA  COD, ATLANTIC        700.0          0.0   
1192  1950          OREGON   COD, PACIFIC      43100.0         20.0   
1193  1950    RHODE ISLAND  COD, ATLANTIC    1226500.0        556.0   
1194  1950        VIRGINIA  COD, ATLANTIC       2600.0          1.0   
1195  1950      WASHINGTON   COD, PACIFIC    7044500.0       3195.0   

          Dollars    Collection      Scientific Name     Tsn      Source  
0     116766003.0    Commercial  Gadus macrocephalus  164711       AKFIN  
1          5031.0    Commercial         Gadus morhua  164712       ACCSP  
2             NaN  Recreational                  NaN  164712        MRIP  
3             NaN  Recreational                  NaN  164712        MRIP  
4             NaN  Recreational                  NaN  164712        MRIP  
...           ...           ...                  ...     ...         ...  
1191         65.0    Commercial         Gadus morhua  164712   ACCSPHIST  
1192       1551.0    Commercial  Gadus macrocephalus  164711  PACFINHIST  
1193      93402.0    Commercial         Gadus morhua  164712   ACCSPHIST  
1194        124.0    Commercial         Gadus morhua  164712   ACCSPHIST  
1195     271116.0    Commercial  Gadus macrocephalus  164711  PACFINHIST  

[1196 rows x 10 columns]

In [4]:
sample.shape

(1196, 10)

##### From the above COD landings dataset, It shows us there are 1196 observations and 10 variables.

In [34]:
sample.head()

Year        State      NMFS Name       Pounds  Metric Tons      Dollars  \
0  2021       ALASKA   COD, PACIFIC  330404171.0     149870.0  116766003.0   
1  2021  CONNECTICUT  COD, ATLANTIC       2277.0          1.0       5031.0   
2  2021  CONNECTICUT  COD, ATLANTIC       7921.0          4.0          NaN   
3  2021     DELAWARE  COD, ATLANTIC         18.0          0.0          NaN   
4  2021        MAINE  COD, ATLANTIC       1171.0          1.0          NaN   

     Collection      Scientific Name     Tsn Source       weight  
0    Commercial  Gadus macrocephalus  164711  AKFIN -330254301.0  
1    Commercial         Gadus morhua  164712  ACCSP      -2276.0  
2  Recreational                  NaN  164712   MRIP      -7917.0  
3  Recreational                  NaN  164712   MRIP        -18.0  
4  Recreational                  NaN  164712   MRIP      -1170.0

##### In the above the COD landings dataset, it shows first five data of the dataset , It starts in the year of 2021.

In [6]:
sample.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1196 entries, 0 to 1195
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Year             1196 non-null   int64  
 1   State            1196 non-null   object 
 2   NMFS Name        1196 non-null   object 
 3   Pounds           1162 non-null   float64
 4   Metric Tons      1162 non-null   float64
 5   Dollars          819 non-null    float64
 6   Collection       1196 non-null   object 
 7   Scientific Name  819 non-null    object 
 8   Tsn              1196 non-null   int64  
 9   Source           1196 non-null   object 
dtypes: float64(3), int64(2), object(5)
memory usage: 93.6+ KB


##### In the COD landings dataset we have three datatypes as float64, int64 and object each of them consisting of 3,2,5 for each data type alomg with memory usage of 93.6 KB. Only variables that are having 819 Non - Null count are Dollars and Scientific name.

In [36]:
sample.columns

Index(['Year', 'State', 'NMFS Name', 'Pounds', 'Metric Tons', 'Dollars',
       'Collection', 'Scientific Name', 'Tsn', 'Source', 'weight'],
      dtype='object')

#####  In the above Dataset we have selected there are column names consisting of Year, State, NMFS Name, Pounds, Metric Tons, Dollars, Collection, Scientific Name, Tsn, Source, weight.

In [8]:
sample.index

RangeIndex(start=0, stop=1196, step=1)

##### In this data set where the RangeIndex Row starts from 0 and ends at 1195 where steps increasing at a rate of 1 per row. 

In [9]:
sample.sort_values(['State'],ascending=True)

Year       State     NMFS Name       Pounds  Metric Tons      Dollars  \
0     2021      ALASKA  COD, PACIFIC  330404171.0     149870.0  116766003.0   
916   1973      ALASKA  COD, PACIFIC     158600.0         72.0      36500.0   
629   1990      ALASKA  COD, PACIFIC  523768974.0     237580.0   91769437.0   
928   1972      ALASKA  COD, PACIFIC      97800.0         44.0      17233.0   
611   1991      ALASKA  COD, PACIFIC  507895347.0     230380.0  113588684.0   
...    ...         ...           ...          ...          ...          ...   
842   1980  WASHINGTON  COD, PACIFIC   12754843.0       5786.0    2394208.0   
117   2016  WASHINGTON  COD, PACIFIC       1246.0          1.0          NaN   
855   1979  WASHINGTON  COD, PACIFIC    7578354.0       3438.0    1591410.0   
788   1983  WASHINGTON  COD, PACIFIC     407280.0        185.0          NaN   
1195  1950  WASHINGTON  COD, PACIFIC    7044500.0       3195.0     271116.0   

        Collection      Scientific Name     Tsn      Source  
0       Commercial  Gadus macrocephalus  164711       AKFIN  
916     Commercial  Gadus macrocephalus  164711   AKFINHIST  
629     Commercial  Gadus macrocephalus  164711   AKFINHIST  
928     Commercial  Gadus macrocephalus  164711   AKFINHIST  
611     Commercial  Gadus macrocephalus  164711   AKFINHIST  
...            ...                  ...     ...         ...  
842     Commercial  Gadus macrocephalus  164711  PACFINHIST  
117   Recreational                  NaN  164711        MRIP  
855     Commercial  Gadus macrocephalus  164711  PACFINHIST  
788   Recreational                  NaN  164711        MRIP  
1195    Commercial  Gadus macrocephalus  164711  PACFINHIST  

[1196 rows x 10 columns]

##### In the above data set we have sorted the values with state variable where it arranged the data in an ascending order in an alphabetical order.

In [10]:
sample['Dollars'].value_counts()

85.0        3
2637.0      2
7.0         2
21.0        2
17.0        2
           ..
2301.0      1
256270.0    1
71.0        1
68994.0     1
271116.0    1
Name: Dollars, Length: 801, dtype: int64

##### In this data we have taken dollars to use this function it show us length of the variable which is 801 and data type is Int64.

In [11]:
sample.drop_duplicates(inplace=True)

##### If there is any duplicates present in the data set it would shows us by having change in the observations and variables, After using this there was no change in the Dataset so there are no duplicates in the dataset

In [12]:
sample

Year           State      NMFS Name       Pounds  Metric Tons  \
0     2021          ALASKA   COD, PACIFIC  330404171.0     149870.0   
1     2021     CONNECTICUT  COD, ATLANTIC       2277.0          1.0   
2     2021     CONNECTICUT  COD, ATLANTIC       7921.0          4.0   
3     2021        DELAWARE  COD, ATLANTIC         18.0          0.0   
4     2021           MAINE  COD, ATLANTIC       1171.0          1.0   
...    ...             ...            ...          ...          ...   
1191  1950  NORTH CAROLINA  COD, ATLANTIC        700.0          0.0   
1192  1950          OREGON   COD, PACIFIC      43100.0         20.0   
1193  1950    RHODE ISLAND  COD, ATLANTIC    1226500.0        556.0   
1194  1950        VIRGINIA  COD, ATLANTIC       2600.0          1.0   
1195  1950      WASHINGTON   COD, PACIFIC    7044500.0       3195.0   

          Dollars    Collection      Scientific Name     Tsn      Source  
0     116766003.0    Commercial  Gadus macrocephalus  164711       AKFIN  
1          5031.0    Commercial         Gadus morhua  164712       ACCSP  
2             NaN  Recreational                  NaN  164712        MRIP  
3             NaN  Recreational                  NaN  164712        MRIP  
4             NaN  Recreational                  NaN  164712        MRIP  
...           ...           ...                  ...     ...         ...  
1191         65.0    Commercial         Gadus morhua  164712   ACCSPHIST  
1192       1551.0    Commercial  Gadus macrocephalus  164711  PACFINHIST  
1193      93402.0    Commercial         Gadus morhua  164712   ACCSPHIST  
1194        124.0    Commercial         Gadus morhua  164712   ACCSPHIST  
1195     271116.0    Commercial  Gadus macrocephalus  164711  PACFINHIST  

[1196 rows x 10 columns]

In [13]:
sample_subset=sample[['State','Collection']]

##### In the above dataset to display state and collection as subset , To display a specific set of data

In [14]:
sample_subset

State    Collection
0             ALASKA    Commercial
1        CONNECTICUT    Commercial
2        CONNECTICUT  Recreational
3           DELAWARE  Recreational
4              MAINE  Recreational
...              ...           ...
1191  NORTH CAROLINA    Commercial
1192          OREGON    Commercial
1193    RHODE ISLAND    Commercial
1194        VIRGINIA    Commercial
1195      WASHINGTON    Commercial

[1196 rows x 2 columns]

In [15]:
sample_row=sample[sample['Metric Tons']>100000]

In [38]:
sample_row

Year   State     NMFS Name       Pounds  Metric Tons      Dollars  \
0    2021  ALASKA  COD, PACIFIC  330404171.0     149870.0  116766003.0   
20   2020  ALASKA  COD, PACIFIC  380338622.0     172520.0  166344712.0   
40   2019  ALASKA  COD, PACIFIC  463723847.0     210344.0  118593170.0   
59   2018  ALASKA  COD, PACIFIC  512455848.0     232448.0  238939847.0   
80   2017  ALASKA  COD, PACIFIC  657266257.0     298134.0  194058038.0   
99   2016  ALASKA  COD, PACIFIC  707151529.0     320762.0  170629498.0   
120  2015  ALASKA  COD, PACIFIC  697160182.0     316230.0  174380395.0   
142  2014  ALASKA  COD, PACIFIC  716593813.0     325045.0  153274859.0   
160  2013  ALASKA  COD, PACIFIC  681318098.0     309044.0  156129081.0   
181  2012  ALASKA  COD, PACIFIC  716899276.0     325183.0  171206230.0   
203  2011  ALASKA  COD, PACIFIC  663188251.0     300820.0  163440122.0   
227  2010  ALASKA  COD, PACIFIC  538759655.0     244380.0  145907806.0   
250  2009  ALASKA  COD, PACIFIC  491072879.0     222749.0   98506890.0   
270  2008  ALASKA  COD, PACIFIC  493968564.0     224063.0  281562082.0   
291  2007  ALASKA  COD, PACIFIC  488536540.0     221599.0  224726808.0   
307  2006  ALASKA  COD, PACIFIC  517799333.0     234872.0  196763747.0   
328  2005  ALASKA  COD, PACIFIC  546748030.0     248003.0  149808960.0   
345  2004  ALASKA  COD, PACIFIC  587337007.0     266414.0  147421589.0   
366  2003  ALASKA  COD, PACIFIC  564561894.0     256084.0  158077330.0   
386  2002  ALASKA  COD, PACIFIC  510758682.0     231679.0   95157191.0   
402  2001  ALASKA  COD, PACIFIC  470768496.0     213539.0  117604636.0   
422  2000  ALASKA  COD, PACIFIC  529663924.0     240254.0  141941773.0   
442  1999  ALASKA  COD, PACIFIC  523280870.0     237359.0   82925077.0   
463  1998  ALASKA  COD, PACIFIC  588271827.0     266838.0  111816350.0   
483  1997  ALASKA  COD, PACIFIC  696452648.0     315909.0  147039622.0   
507  1996  ALASKA  COD, PACIFIC  612275884.0     277727.0  116280895.0   
528  1995  ALASKA  COD, PACIFIC  597760115.0     271142.0  114342440.0   
547  1994  ALASKA  COD, PACIFIC  459169607.0     208278.0   82302096.0   
569  1993  ALASKA  COD, PACIFIC  425237935.0     192887.0   72918495.0   
592  1992  ALASKA  COD, PACIFIC  552594158.0     250655.0  125313941.0   
611  1991  ALASKA  COD, PACIFIC  507895347.0     230380.0  113588684.0   
629  1990  ALASKA  COD, PACIFIC  523768974.0     237580.0   91769437.0   
648  1989  ALASKA  COD, PACIFIC  364722691.0     165437.0   55273590.0   
667  1988  ALASKA  COD, PACIFIC  260281201.0     118063.0   36914772.0   

     Collection      Scientific Name     Tsn     Source  
0    Commercial  Gadus macrocephalus  164711      AKFIN  
20   Commercial  Gadus macrocephalus  164711      AKFIN  
40   Commercial  Gadus macrocephalus  164711      AKFIN  
59   Commercial  Gadus macrocephalus  164711      AKFIN  
80   Commercial  Gadus macrocephalus  164711      AKFIN  
99   Commercial  Gadus macrocephalus  164711      AKFIN  
120  Commercial  Gadus macrocephalus  164711      AKFIN  
142  Commercial  Gadus macrocephalus  164711      AKFIN  
160  Commercial  Gadus macrocephalus  164711      AKFIN  
181  Commercial  Gadus macrocephalus  164711      AKFIN  
203  Commercial  Gadus macrocephalus  164711      AKFIN  
227  Commercial  Gadus macrocephalus  164711      AKFIN  
250  Commercial  Gadus macrocephalus  164711      AKFIN  
270  Commercial  Gadus macrocephalus  164711      AKFIN  
291  Commercial  Gadus macrocephalus  164711      AKFIN  
307  Commercial  Gadus macrocephalus  164711  AKFINHIST  
328  Commercial  Gadus macrocephalus  164711  AKFINHIST  
345  Commercial  Gadus macrocephalus  164711  AKFINHIST  
366  Commercial  Gadus macrocephalus  164711  AKFINHIST  
386  Commercial  Gadus macrocephalus  164711  AKFINHIST  
402  Commercial  Gadus macrocephalus  164711  AKFINHIST  
422  Commercial  Gadus macrocephalus  164711  AKFINHIST  
442  Commercial  Gadus macrocephalus  164711  AKFINHIST  
463  Commercial  Gadus

##### In the above data we have showed values which are greater than 100000 in the metric tons variable. 

In [17]:
sample.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1196 entries, 0 to 1195
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Year             1196 non-null   int64  
 1   State            1196 non-null   object 
 2   NMFS Name        1196 non-null   object 
 3   Pounds           1162 non-null   float64
 4   Metric Tons      1162 non-null   float64
 5   Dollars          819 non-null    float64
 6   Collection       1196 non-null   object 
 7   Scientific Name  819 non-null    object 
 8   Tsn              1196 non-null   int64  
 9   Source           1196 non-null   object 
dtypes: float64(3), int64(2), object(5)
memory usage: 102.8+ KB


##### In the above dataset, After selecting the metric tons values which are greater than 100000 the memory usage has been increased from 93.6 KB to 102.8 KB.

In [18]:
sample['weight']=sample['Metric Tons']-sample['Pounds']

In [19]:
sample['weight']

0      -330254301.0
1           -2276.0
2           -7917.0
3             -18.0
4           -1170.0
           ...     
1191         -700.0
1192       -43080.0
1193     -1225944.0
1194        -2599.0
1195     -7041305.0
Name: weight, Length: 1196, dtype: float64

##### In this above dataset we have added a new variable of weight by using the two existing variables Metric tons and pounds to create an new one.

In [20]:
sample_index=sample.set_index('Year')

In [21]:
sample_index

State      NMFS Name       Pounds  Metric Tons      Dollars  \
Year                                                                         
2021          ALASKA   COD, PACIFIC  330404171.0     149870.0  116766003.0   
2021     CONNECTICUT  COD, ATLANTIC       2277.0          1.0       5031.0   
2021     CONNECTICUT  COD, ATLANTIC       7921.0          4.0          NaN   
2021        DELAWARE  COD, ATLANTIC         18.0          0.0          NaN   
2021           MAINE  COD, ATLANTIC       1171.0          1.0          NaN   
...              ...            ...          ...          ...          ...   
1950  NORTH CAROLINA  COD, ATLANTIC        700.0          0.0         65.0   
1950          OREGON   COD, PACIFIC      43100.0         20.0       1551.0   
1950    RHODE ISLAND  COD, ATLANTIC    1226500.0        556.0      93402.0   
1950        VIRGINIA  COD, ATLANTIC       2600.0          1.0        124.0   
1950      WASHINGTON   COD, PACIFIC    7044500.0       3195.0     271116.0   

        Collection      Scientific Name     Tsn      Source       weight  
Year                                                                      
2021    Commercial  Gadus macrocephalus  164711       AKFIN -330254301.0  
2021    Commercial         Gadus morhua  164712       ACCSP      -2276.0  
2021  Recreational                  NaN  164712        MRIP      -7917.0  
2021  Recreational                  NaN  164712        MRIP        -18.0  
2021  Recreational                  NaN  164712        MRIP      -1170.0  
...            ...                  ...     ...         ...          ...  
1950    Commercial         Gadus morhua  164712   ACCSPHIST       -700.0  
1950    Commercial  Gadus macrocephalus  164711  PACFINHIST     -43080.0  
1950    Commercial         Gadus morhua  164712   ACCSPHIST   -1225944.0  
1950    Commercial         Gadus morhua  164712   ACCSPHIST      -2599.0  
1950    Commercial  Gadus macrocephalus  164711  PACFINHIST   -7041305.0  

[1196 rows x 10 columns]

#####  we have assigned one more column value in the row index ,in the dataset we have applied this function 'year' variable for it. 

In [22]:
sample_loc=sample.loc[(sample.Year>=2020)&(sample.Collection=='Commercial')]

In [23]:
sample_loc

Year          State      NMFS Name       Pounds  Metric Tons      Dollars  \
0   2021         ALASKA   COD, PACIFIC  330404171.0     149870.0  116766003.0   
1   2021    CONNECTICUT  COD, ATLANTIC       2277.0          1.0       5031.0   
5   2021          MAINE  COD, ATLANTIC      47311.0         21.0     128833.0   
6   2021  MASSACHUSETTS  COD, ATLANTIC    1201132.0        545.0    2619925.0   
9   2021  NEW HAMPSHIRE  COD, ATLANTIC      45836.0         21.0     124631.0   
11  2021       NEW YORK  COD, ATLANTIC       3954.0          2.0      12245.0   
13  2021         OREGON   COD, PACIFIC       2205.0          1.0        781.0   
16  2021   RHODE ISLAND  COD, ATLANTIC      12803.0          6.0      26286.0   
18  2021     WASHINGTON   COD, PACIFIC      69446.0         32.0      37853.0   
20  2020         ALASKA   COD, PACIFIC  380338622.0     172520.0  166344712.0   
21  2020    CONNECTICUT  COD, ATLANTIC       4492.0          2.0       7735.0   
24  2020          MAINE  COD, ATLANTIC      58733.0         27.0     149844.0   
27  2020  MASSACHUSETTS  COD, ATLANTIC    1468781.0        666.0    3184861.0   
29  2020  NEW HAMPSHIRE  COD, ATLANTIC      67340.0         31.0     182664.0   
31  2020       NEW YORK  COD, ATLANTIC       5590.0          3.0      15940.0   
33  2020         OREGON   COD, PACIFIC       6173.0          3.0       2637.0   
36  2020   RHODE ISLAND  COD, ATLANTIC      17589.0          8.0      37426.0   
38  2020     WASHINGTON   COD, PACIFIC     179236.0         81.0      94776.0   

    Collection      Scientific Name     Tsn  Source       weight  
0   Commercial  Gadus macrocephalus  164711   AKFIN -330254301.0  
1   Commercial         Gadus morhua  164712   ACCSP      -2276.0  
5   Commercial         Gadus morhua  164712   ACCSP     -47290.0  
6   Commercial         Gadus morhua  164712   ACCSP   -1200587.0  
9   Commercial         Gadus morhua  164712   ACCSP     -45815.0  
11  Commercial         Gadus morhua  164712   ACCSP      -3952.0  
13  Commercial  Gadus macrocephalus  164711  PACFIN      -2204.0  
16  Commercial         Gadus morhua  164712   ACCSP     -12797.0  
18  Commercial  Gadus macrocephalus  164711  PACFIN     -69414.0  
20  Commercial  Gadus macrocephalus  164711   AKFIN -380166102.0  
21  Commercial         Gadus morhua  164712   ACCSP      -4490.0  
24  Commercial         Gadus morhua  164712   ACCSP     -58706.0  
27  Commercial         Gadus morhua  164712   ACCSP   -1468115.0  
29  Commercial         Gadus morhua  164712   ACCSP     -67309.0  
31  Commercial         Gadus morhua  164712   ACCSP      -5587.0  
33  Commercial  Gadus macrocephalus  164711  PACFIN      -6170.0  
36  Commercial         Gadus morhua  164712   ACCSP     -17581.0  
38  Commercial  Gadus macrocephalus  164711  PACFIN    -179155.0

##### In this dataset we have used this function to locate the specific data which has the value called commercial.

In [24]:
sample_iloc=sample.iloc[1:3,2:4]

In [25]:
sample_iloc

NMFS Name  Pounds
1  COD, ATLANTIC  2277.0
2  COD, ATLANTIC  7921.0

##### we have displayed the specific rows and columns of [1:3 ,2:4] in order to display the specific data in the dataset.

In [26]:
sample_agg=sample.groupby('Dollars')['Tsn'].mean()

In [27]:
sample_agg

Dollars
1.0            164712.0
3.0            164712.0
4.0            164712.0
7.0            164712.0
8.0            164711.0
                 ...   
194058038.0    164711.0
196763747.0    164711.0
224726808.0    164711.0
238939847.0    164711.0
281562082.0    164711.0
Name: Tsn, Length: 801, dtype: float64

#####  In this we have used variable named dollars and Tsn which variables are grouped and calculated their mean value with the Tsn value of length 801, data type as float64. 

In [28]:
sample['Tsn']=sample['Tsn'].astype('object')

In [29]:
sample.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1196 entries, 0 to 1195
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Year             1196 non-null   int64  
 1   State            1196 non-null   object 
 2   NMFS Name        1196 non-null   object 
 3   Pounds           1162 non-null   float64
 4   Metric Tons      1162 non-null   float64
 5   Dollars          819 non-null    float64
 6   Collection       1196 non-null   object 
 7   Scientific Name  819 non-null    object 
 8   Tsn              1196 non-null   object 
 9   Source           1196 non-null   object 
 10  weight           1162 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 112.1+ KB


##### we have use .astype function to change datatype of a variable. In this we have changed Tsn as object type where it was a int64 before using this function.

In [30]:
sample.tail()

Year           State      NMFS Name     Pounds  Metric Tons   Dollars  \
1191  1950  NORTH CAROLINA  COD, ATLANTIC      700.0          0.0      65.0   
1192  1950          OREGON   COD, PACIFIC    43100.0         20.0    1551.0   
1193  1950    RHODE ISLAND  COD, ATLANTIC  1226500.0        556.0   93402.0   
1194  1950        VIRGINIA  COD, ATLANTIC     2600.0          1.0     124.0   
1195  1950      WASHINGTON   COD, PACIFIC  7044500.0       3195.0  271116.0   

      Collection      Scientific Name     Tsn      Source     weight  
1191  Commercial         Gadus morhua  164712   ACCSPHIST     -700.0  
1192  Commercial  Gadus macrocephalus  164711  PACFINHIST   -43080.0  
1193  Commercial         Gadus morhua  164712   ACCSPHIST -1225944.0  
1194  Commercial         Gadus morhua  164712   ACCSPHIST    -2599.0  
1195  Commercial  Gadus macrocephalus  164711  PACFINHIST -7041305.0

##### In the above the COD landings dataset, it shows last five data of the dataset , It starts in the year of 1950

In [31]:
sample.describe()

Year        Pounds    Metric Tons       Dollars        weight
count  1196.000000  1.162000e+03    1162.000000  8.190000e+02  1.162000e+03
mean   1990.091137  1.966439e+07    8919.712565  7.660947e+06 -1.965547e+07
std      19.794997  9.329291e+07   42317.391759  2.984627e+07  9.325060e+07
min    1950.000000  2.000000e+00       0.000000  1.000000e+00 -7.165741e+08
25%    1975.000000  1.502450e+04       7.000000  9.271000e+03 -2.147429e+06
50%    1992.000000  3.082365e+05     139.500000  1.352060e+05 -3.080970e+05
75%    2007.000000  2.148404e+06     974.750000  1.143642e+06 -1.501750e+04
max    2021.000000  7.168993e+08  325183.000000  2.815621e+08 -2.000000e+00

##### To perform statistical functions for the dataset ,we use sample describe function, like count, mean, std, min, 25% of value, 50% of value, 75% of value and their maximum.

In [32]:
sample_index=sample.reset_index()

In [37]:
sample.reset_index

<bound method DataFrame.reset_index of       Year           State      NMFS Name       Pounds  Metric Tons  \
0     2021          ALASKA   COD, PACIFIC  330404171.0     149870.0   
1     2021     CONNECTICUT  COD, ATLANTIC       2277.0          1.0   
2     2021     CONNECTICUT  COD, ATLANTIC       7921.0          4.0   
3     2021        DELAWARE  COD, ATLANTIC         18.0          0.0   
4     2021           MAINE  COD, ATLANTIC       1171.0          1.0   
...    ...             ...            ...          ...          ...   
1191  1950  NORTH CAROLINA  COD, ATLANTIC        700.0          0.0   
1192  1950          OREGON   COD, PACIFIC      43100.0         20.0   
1193  1950    RHODE ISLAND  COD, ATLANTIC    1226500.0        556.0   
1194  1950        VIRGINIA  COD, ATLANTIC       2600.0          1.0   
1195  1950      WASHINGTON   COD, PACIFIC    7044500.0       3195.0   

          Dollars    Collection      Scientific Name     Tsn      Source  \
0     116766003.0    Commercial 

#####  In this above dataset, we have set a integers from the row of 0 to 1195 and 11 columns.